spec/models/local_authority_spec.rb

```diff
@@ -76,5 +76,13 @@
         expect(@local_authority.link_last_checked).to be_nil
       end
     end
+
+    describe "calculate_count_of_broken_links" do
+      it "calculates the correct number of broken for the local authority" do
+        local_authority = FactoryGirl.create(:local_authority)
+        FactoryGirl.create(:link, :with_service_interaction, local_authority: local_authority, status: 500)
+        expect { local_authority.calculate_count_of_broken_links }.to change { local_authority.broken_link_count }.from(0).to(1)
```

Comment:
We can't actually get this test to pass. It appears that when the link instance is being created by FactoryGirl, the after_save callback is already being called so when we get to L84 the broken_link_count is already 1, not 0 as currently expected...